docs/architecture/adr/ADR046-replace-harbor-ecr.md

```diff
@@ -0,0 +1,43 @@
+# ADR046: Replacing harbor with ECR
+
+## Status
+
+Accepted
+
+## Context
+
+Harbor has proved itself to be rather unstable (causing several incidents from
+various root causes). We have also been gradually rolling back our usage within
+GSP of the various tools that harbor provides:
+
+* we are not using notary for signing (image provenance); this is now being
+  handled by gatekeeper
+* we haven't yet implemented anything using Clair
+* we no longer have a requirement for public and private repositories
+
+It is also worth noting that some of these (e.g. "Clair") have been integrated
+into AWS ECR since we last looked.
+
+Following the "dev namespaces" work we now have a requirement to secure image
+"push" actions to per-namespace credentials. In harbor this would require
+interactions with the harbor API and doesn't fit very well with "desired state
+configuration".
```

Comment:
This paragraph could be a bit clearer. Maybe something like

> With the "dev namespaces" work we want to ensure each namespace has its own place to push docker images
> and that one namespace can't overwrite another namespace's docker images. This would be very fiddly to achieve
> with harbor because it would require
> interactions with the harbor API and doesn't fit very well with "desired state
> configuration".  However with ECR we can quite easily give each namespace its own ECR repository, which
> would fulfil this requirement.

Reply:
Reworded more like yours ... but i avoided "each namespace gets an ECR repository" since it's more like that it's more suitable for provisioning with seperate creds